include/rcpputils/filesystem_helper.hpp

```diff
@@ -65,14 +66,21 @@ static constexpr const char kPreferredSeparator = RCPPUTILS_IMPL_OS_DIRSEP;
 
 #undef RCPPUTILS_IMPL_OS_DIRSEP
 
-
+// TODO(ahcorde): Remove deprecated class on the next release.
+#if !defined(_WIN32)
+# pragma GCC diagnostic push
+# pragma GCC diagnostic ignored "-Wdeprecated-declarations"
+#else  // !defined(_WIN32)
+# pragma warning(push)
+# pragma warning(disable: 4996)
+#endif
```

Comment:
I put this block here because not only the class is generating the deprecation warnings,  there are some  functions also like `remove_extension`, `create_directories`, etc.